### PR TITLE
HPCC-29421 ALA CurlClient expand err messages

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@ghalliday this PR would back-port a change we only included in 8.12.x+ but it would beneficial for debugging issues ops (8.10.x) encounters, and adds a change on top of it which expands the error message.

**Review by ghalliday — COMMENTED** [review]
@rpastrana change looks good.  One question about the code.  I would merge as-is, but I think it can be simplified.

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 1]
```diff
@@ -247,10 +247,28 @@ static void submitKQLQuery(std::string & readBuffer, const char * token, const c
...
 
-            if (response_code == 404L)
-                throw makeStringExceptionV(-1, "%s KQL request: Error (404): Ensure the WorkspaceID (%s) is valid!", COMPONENT_NAME, workspaceID);
-
-            throw makeStringExceptionV(-1, "%s KQL request: Error (%d): %s", COMPONENT_NAME, curlResponseCode, (curlErrBuffer[0] ? curlErrBuffer : "<unknown>"));
+            StringBuffer message;
+            switch (response_code)
+            {
+            case 400L:
+                throw makeStringExceptionV(-1,"%s KQL response: Error (400): Request is badly formed and failed (permanently): '%s'", COMPONENT_NAME, (curlErrBuffer[0] ? curlErrBuffer : ""));
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:254]: Looks like libcurl initializes the err buffer to empty string, I'll update.

**Comment by rpastrana** [comment]
@ghalliday performed the discussed and change and went ahead a squashed due to nature of the change.
This PR contains 2 commits because [4a6395b65490aeea007b1e9793faaf49fb6c4afb ](https://github.com/hpcc-systems/HPCC-Platform/pull/16927) should have been back ported to 8.10.x, but we only targeted 8.12.x+
This PR is dependent on that change and should target 8.10.x due to internal customer use of that branch.

**Comment by ghalliday** [comment]
@rpastrana I debated including it in 8.10.x - that should really only be getting security fixes.  Given it is a safe change, and it will primarily help our debugging I think it is reasonable.

**Comment by rpastrana** [comment]
Ag

> @rpastrana I debated including it in 8.10.x - that should really only be getting security fixes. Given it is a safe change, and it will primarily help our debugging I think it is reasonable.

Understood and in agreement.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**